qiime2/core/path.py

```diff
@@ -96,10 +94,16 @@ def __new__(cls, dir=False, **kwargs):
         """
         Create a tempfile, return pathlib.Path reference to it.
         """
+        from qiime2.core.cache import get_cache
+
+        cache = get_cache()
+        tmp_path = cache.process_pool.get_tmp_path()
+        prefix = 'q2-%s-' % cls.__name__
+
         if dir:
-            name = tempfile.mkdtemp(**kwargs)
+            name = tempfile.mkdtemp(prefix=prefix, dir=tmp_path)
```

Comment:
I would remove `**kwargs` from the signature of `__new__` if it isn't used anymore.